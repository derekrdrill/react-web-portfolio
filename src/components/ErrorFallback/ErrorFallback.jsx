import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Grid, Typography } from '@mui/material';

import derekBroken from '../../assets/derekBroken.png';

const ErrorFallback = () => {
  return (
    <ErrorFallbackContainer container>
      <Grid item xs={12}>
        <Typography variant='h5' textAlign='center'>
          Uh oh.. looks like something broke ðŸ˜­ðŸ’”
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <Grid container justifyContent='center'>
          <ErrorFallbackImage src={derekBroken} />
        </Grid>
      </Grid>
      <Grid item xs={12}>
        <Typography variant='body1' textAlign='center'>
          This has been logged, and is actively being fixed
        </Typography>
      </Grid>
      <Grid item xs={12}>
        <Typography variant='body1' textAlign='center'>
          Until then, try <a href='/'>going home</a> and continue browsing
        </Typography>
      </Grid>
    </ErrorFallbackContainer>
  );
};

ErrorFallback.propTypes = {
  children: PropTypes.node,
};

export default ErrorFallback;

export const ErrorFallbackContainer = styled(Grid)({
  backgroundColor: 'azure',
  height: '100vh',
  paddingTop: 50,
});

export const ErrorFallbackImage = styled.img({
  borderRadius: 100,
});
