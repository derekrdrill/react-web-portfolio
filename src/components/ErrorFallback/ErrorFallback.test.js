import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ErrorFallback from './ErrorFallback';

const renderer = new ShallowRenderer();

describe('ErrorFallback tests', () => {
  it('renders correctly', () => {
    renderer.render(<ErrorFallback />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
