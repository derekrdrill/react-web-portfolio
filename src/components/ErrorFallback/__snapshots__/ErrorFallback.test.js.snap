// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorFallback tests renders correctly 1`] = `
<Styled(Component)
  container={true}
>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <ForwardRef(Typography)
      textAlign="center"
      variant="h5"
    >
      Oh no! Looks like an error has been encountered!!
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <ForwardRef(Typography)
      textAlign="center"
      variant="body1"
    >
      This has been logged, and is actively being fixed
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <ForwardRef(Typography)
      textAlign="center"
      variant="body1"
    >
      Until then, try 
      <a
        href="/"
      >
        going home
      </a>
       and continue browsing
    </ForwardRef(Typography)>
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    xs={12}
  >
    <ForwardRef(Grid)
      container={true}
      justifyContent="center"
    >
      <styled.img
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</Styled(Component)>
`;
